src/main/java/StudentGroup.java

```diff
@@ -0,0 +1,61 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class StudentGroup {
+
+    private Student groupLeader;
+    private List<Student> students = new ArrayList<>();
+    private final List<String> tasks = new ArrayList<>();
+    Map<List<Student>, String> completedTasks = new HashMap<>();
+
+
+    public StudentGroup(Student groupLeader) {
+        this.groupLeader = groupLeader;
+    }
+
+    public Student changeGroupLeader(Student newGroupLeader) {
+        this.groupLeader = newGroupLeader;
+        return newGroupLeader;
+    }
+
+    public void addStudent(Student student) {
+        if (students.contains(student)) {
+            System.out.println("This student is already exist! You are can't add him.");
+        }
+        this.students.add(student);
+    }
+
+    public void removeStudent(Student student) {
+        if (student == groupLeader) {
+            System.out.println("You are can't remove this student because he is a Group Leader!");
+        } else {
+            this.students.remove(student);
+        }
+    }
+
+    public boolean addGroupTask(String task) {
+        tasks.add(task);
+        return false;
+    }
+
+    public void markTaskAsDone(Student student, String task) {
+        if (students.contains(student) && this.tasks.contains(task)) {
+            completedTasks.put(students, task);
```

Comment:
Previously I left comment that you need to use structure `Map<String, List<Student>> completedTasks = new HashMap<>();` for class field
In this place you need to do next:
```
if (!compleatedTasks.containsKey(task))
   compleatedTasks.put(task, new ArryList<>());
compleatedTasks.get(task).add(student);
```
And that is it